src/components/Authentication/Signup/Signup.jsx

```diff
@@ -0,0 +1,141 @@
+import loginSignupImage from "../../../assets/login-signup.png";
+import "@fontsource/dm-sans";
+import { useFormik } from "formik";
+import * as Yup from "yup";
+import { useNavigate } from "react-router-dom";
+
+function Signup() {
+  const navigate = useNavigate();
+
+  const validationSchema = Yup.object().shape({
+    name: Yup.string()
+      .min(3, "Name must be at least 3 characters")
+      .matches(/^[a-zA-Z]+$/, "Name must contain only letters"),
+    email: Yup.string()
+      .email("Invalid email format")
+      .required("Email is required"),
+    password: Yup.string()
+      .required("Password is required")
+      .matches(
+        /^(?=.*[a-z])(?=.*[A-Z])(?=.*\d)[a-zA-Z\d]{8,}$/,
+        "Invalid password."
+      ),
+  });
```

Comment:
validation schemas are better kept in a different file. If you create them here, they will be recreated with every component re-render.

Reply:
Fixed the problem.